### DISCUSSION TITLE
Difference between c and C when keymapping using SHIFT control modifier

### DISCUSSION THREAD

**Discussion by MartyMcFlyInTheSky** [wezterm/wezterm#5921] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

gnome

### WezTerm version

wezterm 20240730-082727-56a27e93

### Ask your question!

What is the difference between mapping:

```
{
        key = 'c',
        mods = 'CTRL|SHIFT',
        action = ...
}
```

and 

```
{
        key = 'C',
        mods = 'CTRL|SHIFT',
        action = ...
}
```

in the wezterm config?

Since for this keystroke I need to press shift in any case which would make my 'C' uppercase, so how could I ever make use of the lowercase 'c' keybinding? Does it have any effect?

I'm confused because when reading wezterms [default key assignements](https://wezfurlong.org/wezterm/config/default-keys.html), there is an entry for CTRL + SHIFT + t (lowercase) and one for SUPER+SHIFT+T (uppercase). I would have thought that it's all the same, but apparently not.

**Comment by MartyMcFlyInTheSky** [comment]
*push*

**Comment by MartyMcFlyInTheSky** [comment]
push2

**Comment by neogeographica** [comment, marked as answer]
Someone else may know more definitively, but I don't think there is a difference. 

Cf. the comment at https://github.com/wez/wezterm/blob/2d0c5cddc91a9c59aef9a7667d90924e7cedd0ac/wezterm-gui/src/inputmap.rs#L52
```rust
                // If the user configures {key='p', mods='CTRL|SHIFT'} that gets
                // normalized into {key='P', mods='CTRL'} in Config::key_bindings(),
                // and that value exists in `keys.default` when we reach this point.
```

**Reply by MartyMcFlyInTheSky** [reply]
Oh very insightful, thank you!
